src/Utility/Utils.cs

```diff
@@ -1,102 +1,131 @@
-                    // Try converting the value to HttpResponseContext if it's not already an object of such type.
-                    if (value is HttpResponseContext)
-                    {
-                        break;
-                    }
-
-                    try
-                    {
-                        value = LanguagePrimitives.ConvertTo<HttpResponseContext>(value);
-                    }
-                    catch (PSInvalidCastException ex)
-                    {
-                        string errorMsg = string.Format(PowerShellWorkerStrings.FailToConvertToHttpResponseContext, bindingName, ex.Message);
-                        throw new InvalidOperationException(errorMsg);
-                    }
-
-                    break;
-                default:
-                    break;
-            }
-
-            return value;
-        }
-    }
-}
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.IO;
+using System.Management.Automation;
+using System.Management.Automation.Runspaces;
+using System.Text;
+using Microsoft.PowerShell.Commands;
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.Utility
+{
+    internal class Utils
+    {
+        internal static CmdletInfo ImportModuleCmdletInfo = new CmdletInfo("Import-Module", typeof(ImportModuleCommand));
+        internal static CmdletInfo RemoveModuleCmdletInfo = new CmdletInfo("Remove-Module", typeof(RemoveModuleCommand));
+        internal static CmdletInfo GetJobCmdletInfo = new CmdletInfo("Get-Job", typeof(GetJobCommand));
+        internal static CmdletInfo RemoveJobCmdletInfo = new CmdletInfo("Remove-Job", typeof(RemoveJobCommand));
+
+        private static InitialSessionState s_initialSessionState = null;
+
+        internal static InitialSessionState GetInitialSessionState()
+        {
+            if (s_initialSessionState == null)
+            {
+                var iss = InitialSessionState.CreateDefault();
+                iss.ThreadOptions = PSThreadOptions.UseCurrentThread;
+                iss.EnvironmentVariables.Add(
+                    new SessionStateVariableEntry(
+                        "PSModulePath",
+                        FunctionLoader.FunctionModulePath,
+                        description: null));
+
+                // Setting the execution policy on macOS and Linux throws an exception so only update it on Windows
+                if(Platform.IsWindows)
+                {
+                    // This sets the execution policy on Windows to Unrestricted which is required to run the user's function scripts on
+                    // Windows client versions. This is needed if a user is testing their function locally with the func CLI.
+                    iss.ExecutionPolicy = Microsoft.PowerShell.ExecutionPolicy.Unrestricted;
+                }
+
+                s_initialSessionState = iss;
```

Comment:
Is reusing same instance of InitialSessionState not going to cause issue especially when PowerShell instance pool size is greater than 1, say for example if an instance of PowerShell goes wrong then during Dispose associated session will be disposed as well which will cause other PowerShell instance not function correctly?

Reply:
I'm glad you asked this question!

A cloned copy of the passed in `initialsessionstate` will be actually used for the Runspace creation, see https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/hostifaces/ConnectionBase.cs#L78
So it's safe to reuse the same `initialsessionstate` to create multiple Runspaces.